src/Components/Subscriptions/index.js

```diff
@@ -1,9 +1,253 @@
+        method: 'DELETE'
+      });
+      const data = await response.json();
+      setSubscriptions(subscriptions.filter((subscription) => subscription._id !== id));
+      return data;
+    } catch (error) {
+      console.error;
+    }
+  };
+
+  const getSubscriptions = async () => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API}/api/subscriptions/all`);
+      const { data } = await res.json();
+      setSubscriptions(data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const getMembers = async () => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API}/api/members/`);
+      const { data } = await res.json();
+      setMembers(data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const getClasses = async () => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API}/api/classes/`);
+      const { data } = await res.json();
+      setClasses(data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const onClick = async () => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API}/api/subscriptions/`, {
+        method: 'POST',
+        body: JSON.stringify({
+          classId: classId,
+          memberId: memberId,
+          date: date
+        }),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await res.json();
+      console.log('prueba data', data);
+
+      const newSubscription = {
+        _id: data.data._id,
+        classId: {
+          activityId: {
+            name: classes.find((item) => item._id === data.data.classId).activityId?.name
+          }
+        },
+        memberId: {
+          lastName: members.find((item) => item._id === data.data.memberId).lastName
+        },
+        date: data.data.date
+      };
+
+      setSubscriptions([...subscriptions, newSubscription]);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    onClick();
+    setClassId('');
+    setMemberId('');
+  };
+
+  const onEdit = async (id) => {
+    const index = subscriptions.findIndex((subscription) => subscription._id === id);
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API}/api/subscriptions/${id}`, {
+        method: 'PUT',
+        body: JSON.stringify({
+          classId: classId,
+          memberId: memberId,
+          date: date
+        }),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await res.json();
+      console.log('prueba data update', data);
+
+      const updateSubscription = [...subscriptions];
+      updateSubscription[index] = data.data;
+      console.log(updateSubscription[index]);
+
+      setSubscriptions(updateSubscription);
+    } catch (error) {
+      console.error;
```

Comment:
this should be console.error(error); as well

Reply:
No, console.error it's ok